### PR TITLE
Composability interactor refactor

### PR DESCRIPTION
**Pull request by CostinCarabas**
Split State logic in state and config

### CODE REVIEW

File: contracts/feature-tests/composability/interact/src/composability_interact_config.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+use serde::Deserialize;
+use std::io::Read;
+
+/// Config file
+const CONFIG_FILE: &str = "config.toml";
+
+/// Multisig Interact configuration
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    gateway: String,
```
Comment (ovstinga) [inline contracts/feature-tests/composability/interact/src/composability_interact_config.rs:10]: according to the `config.toml`, there is no `gateway`, the deserialization will not go through

File: contracts/feature-tests/composability/interact/src/composability_interact_state.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    pub fn set_forwarder_raw_address(&mut self, address: &str) {
+        self.forwarder_raw_address = Some(String::from(address));
+    }
+    /// Sets the promises address
+    pub fn set_promises_address(&mut self, address: &str) {
+        self.promises_address = Some(String::from(address));
+    }
+
+    /// Returns the vault contract
+    pub fn _vault(&self) -> VaultContract {
```
Comment (ovstinga) [inline contracts/feature-tests/composability/interact/src/composability_interact_state.rs:54]: no need to prefix method name with '_'

Comment (CostinCarabas) [reply contracts/feature-tests/composability/interact/src/composability_interact_state.rs:54]: I don't use this method and I used `_` to silence clippy. Will remove it when I call the method.

File: contracts/feature-tests/composability/interact/src/composability_interact_config.rs (hunk 1 above)
Comment (CostinCarabas) [reply contracts/feature-tests/composability/interact/src/composability_interact_config.rs:10]: Fixed, thanks!

File: contracts/feature-tests/composability/interact/.gitignore [hunk 3]
```diff
@@ -8,3 +8,4 @@
 
 # Temporary storage of deployed contract address, so we can preserve the context between executions.
 composability_address.txt
```
Comment (ovstinga) [inline contracts/feature-tests/composability/interact/.gitignore:10]: old way of storing can be removed

**Review by ovstinga — APPROVED** [review]

File: contracts/feature-tests/composability/interact/src/composability_interact_state.rs (hunk 2 above)
Comment (andrei-marinica) [reply contracts/feature-tests/composability/interact/src/composability_interact_state.rs:54]: You can also do `#[allow(dead_code)]`, gets rid of all warnings

**Review by andrei-marinica — APPROVED** [review]

**Review by alyn509 — APPROVED** [review]

**Pull request merged**